source/extensions/filters/listener/http_inspector/http_inspector.cc

```diff
@@ -23,13 +23,15 @@ Config::Config(Stats::Scope& scope)
 
 const absl::string_view Filter::HTTP2_CONNECTION_PREFACE = "PRI * HTTP/2.0\r\n\r\nSM\r\n\r\n";
 
-Filter::Filter(const ConfigSharedPtr config) : config_(config), no_op_callbacks_() {
+Filter::Filter(const ConfigSharedPtr config)
+    : config_(config), no_op_callbacks_(),
+      requested_read_bytes_(Config::DEFAULT_INITIAL_BUFFER_SIZE) {
   // Filter for only Request Message types with NoOp Parser callbacks.
   if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.http_inspector_use_balsa_parser")) {
     // Set both allow_custom_methods and enable_trailers to true with BalsaParser.
     parser_ = std::make_unique<Http::Http1::BalsaParser>(
-        Http::Http1::MessageType::Request, &no_op_callbacks_, max_request_headers_kb_ * 1024, true,
-        true);
+        Http::Http1::MessageType::Request, &no_op_callbacks_,
+        Config::MAX_INSPECT_SIZE + max_request_headers_kb_, true, true);
```

Comment:
is this change necessary?

Reply:
Probably not with the removal of the configurability.